sandbox.go

```diff
@@ -691,6 +691,9 @@ func (sb *sandbox) EnableService() error {
 func (sb *sandbox) DisableService() error {
 	logrus.Debugf("DisableService %s START", sb.containerID)
 	for _, ep := range sb.getConnectedEndpoints() {
+		if err := ep.deleteServiceInfoFromCluster(sb, "DisableService"); err != nil {
```

Comment:
@fcrisciani That explains why the [moby #31142](https://github.com/moby/moby/pull/31142) patch. doesn't help in 17.06. 

A valid use case shouldn't be broken to fix the race. In this case we can avoid the race by checking the `serviceEnabled` flag. IOW, `deleteServiceInfoFromCluster` will be called only once for one endpoint.